File: 2-pattrons-de-structures/adapter/code-typescript/round/RoundHole.ts
import { RoundPeg } from "./RoundPeg.js";

export class RoundHole {
  constructor(private raduis: number) {}

  get getRaduis(): number {
    return this.raduis;
  }

  public fits(peg: RoundPeg): boolean {
    return this.getRaduis >= peg.getRaduis;
  }
}

File: 2-pattrons-de-structures/adapter/code-java/round/RoundHole.java
package round;

import round.RoundPeg;

public class RoundHole {
    private double radius;

    public RoundHole(double radius){
        this.radius = radius;
    }

    public double getRaduis(){
        return radius;
    }

    public boolean fits(RoundPeg peg) {
        boolean awnser;
        
        awnser = this.getRaduis() >= peg.getRaduis();

        return awnser;
    }
}
